### PR TITLE
Added tests for model checkpoints including volatile

### PR DESCRIPTION
**Pull request by Raalsky**

### CODE REVIEW

File: e2e_tests/integrations/test_huggingface.py [hunk 1]
```diff
@@ -461,3 +463,173 @@ def test_usages(self):
         assert "NeptuneCallback" in [
             type(callback).__name__ for callback in trainer.callback_handler.callbacks
         ]
+
+    def test_model_checkpoints_same(self, environment):
+        with init_run(project=environment.project, api_token=environment.user_token) as run:
+            run_id = run["sys/id"].fetch()
+            callback = NeptuneCallback(run=run, log_checkpoints="same")
+            training_args = self._trainer_default_attributes.copy()
```
Comment (shnela) [inline e2e_tests/integrations/test_huggingface.py:471]: It's inconsistent with fixed tests version where we don't use `copy()` in `self._trainer_default_attributes.copy()`.

Here and in several other places.

File: e2e_tests/integrations/test_huggingface.py [hunk 2]
```diff
@@ -461,3 +463,173 @@ def test_usages(self):
...
+            trainer = Trainer(**training_args, callbacks=[callback])
+            trainer.train(resume_from_checkpoint="model/checkpoint-2000")
+
+        run = init_run(
+            run=run_id,
+            project=environment.project,
+            api_token=environment.user_token,
+            mode="read-only",
+        )
+        assert run["finetuning/train/epoch"].fetch_last() == 1000
```
Comment (shnela) [inline e2e_tests/integrations/test_huggingface.py:513]: I think that we should use `_test_with_run_initialization` twice in this test.

Comment (shnela) [reply e2e_tests/integrations/test_huggingface.py:513]: The same is true for other tests.

**Review by shnela — CHANGES_REQUESTED** [review]

File: e2e_tests/integrations/test_huggingface.py (hunk 1 above)
Comment (Raalsky) [reply e2e_tests/integrations/test_huggingface.py]: Done

File: e2e_tests/integrations/test_huggingface.py (hunk 2 above)
Comment (Raalsky) [reply e2e_tests/integrations/test_huggingface.py]: Done

**Review by shnela — APPROVED** [review]

**Review by shnela — APPROVED** [review]

**Pull request merged**